shared/modules/transaction.utils.js

```diff
@@ -131,3 +134,47 @@ export async function determineTransactionType(txParams, query) {
 
   return { type: result, getCodeResponse: contractCode };
 }
+
+export async function determineTransactionAssetType(
+  txMeta,
+  query,
+  getTokenStandardAndDetails,
+) {
+  // Because we will deal with all types of transactions (including swaps)
+  // we want to get an inferrable type of transaction that isn't special cased
+  // that way we can narrow the number of logic gates required.
+  const { type: inferrableType } = await determineTransactionType(
+    txMeta.txParams,
+    query,
+  );
```

Comment:
@dan437 @danjm -- should we do something else for swaps to determine the asset_type? If we are recording the asset_type, how can we get the asset the user is swapping *from*